### ISSUE TITLE
Log duplicate target paths

### ISSUE THREAD

**Issue by timbunce** [gohugoio/hugo#3113]
I'm using 0.19 now but I'm pretty sure I saw this with 0.18.x as well.

Our layouts/partials/category_nav.html file looks like this:
```
{{ if .IsPage }}
    {{ $.Scratch.Set "currentRelPermalink" .RelPermalink }}
    {{ $.Scratch.Set "currentCategory" "" }}
    {{ if .Params.categories -}}
        {{ $.Scratch.Set "currentCategory" (index .Params.categories 0) }}
    {{ end }}
{{ else }}
    {{ $.Scratch.Set "currentRelPermalink" .URL }}
    {{ $.Scratch.Set "currentCategory" .Title }}
{{ end }}
{{ $currentRelPermalink := $.Scratch.Get "currentRelPermalink" }}
{{ $currentCategory := $.Scratch.Get "currentCategory" | urlize }}

<nav class="category-nav">
    <div class="pages-in-category">
        <strong>{{ if .IsPage }}Other p{{ else }}P{{ end }}ages in {{ partial "category_name.html" $currentCategory }}</strong>
        <ul>
        {{ $category := index .Site.Taxonomies.categories $currentCategory }}
        {{ range $page := $category }}
            {{ if ne $currentRelPermalink .Page.RelPermalink }}
            <li><a href="{{ .Page.RelPermalink }}">{{ .Page.Title }}</a></li>
            {{ end }}
        {{ end }}
        </ul>
    </div>

    <div class="other-categories">
        <strong>Other categories</strong>
        <ul>
        {{ range $cName, $cData := .Site.Taxonomies.categories }}
            {{ if ne $currentCategory $cName }}
            <li><a href="/categories/{{ $cName | urlize }}/">{{ partial "category_name.html" $cName }}</a></li>
            {{ end }}
        {{ end }}
        </ul>
    </div>
</nav>
```

Note the `{{ if .IsPage }}Other p{{ else }}P{{ end }}`.

When I run `hugo` multiple times making to changes to anything, sometimes the `{{ if .IsPage }}` is true and sometimes it's false.

I noticed this because I checked in the public/ directory to git so I could see the impact of changes.

Tim.

**Comment by bep** [comment]
I'm pretty sure what you see is not a race.

My best guess, without seeing the full source, is that you have some kind of conflicting target paths, say two /section/index.html files. Since Hugo 0.18, Hugo renders "everything" in parallel, so these mistakes have become more visible.

If you still feel that this is an issue/bug, please post a link to the full source.

**Comment by timbunce** [comment]
Ah! I see the source file is called `syslog/index.md`. I've renamed it and the output is stable again. Thanks!

Rather than _silent_ instability and corruption, could hugo detect and report this somehow?
Perhaps by keeping track of files written during a build and reporting an error if a file is written more than once?

**Comment by bep** [comment]
>Rather than silent instability and corruption, could hugo detect and report this somehow?

We should do that. You are not the only one who have experienced similar.

**Comment by bep** [comment]
Also note that since hugo 0.19 you have a `disableKinds` option where you can turn off "conflicting" page types, so to speak, or stuff that you don't want/need.

**Comment by max-arnold** [comment]
My workaround https://discourse.gohugo.io/t/can-hugo-fail-on-duplicate-pages-files-in-the-public-dir/14172:

```
{{ range .Site.AllPages.GroupBy "Permalink" }}
  {{ if gt (len .Pages) 1 }}
    {{ errorf "Multiple pages with the same url %q:" .Key }}
    {{ range .Pages }}
      {{ errorf "\t %q" .File.Path }}
    {{ end }}
    {{ .DuplicatePageUrlsFound }}
  {{ end }}
{{ end }}
```

Remove the `{{ .DuplicatePageUrlsFound }}` if you only want warnings.

**Comment by bep** [comment]
I have added a patch for this (I needed this for my own debugging), but I want to have it hide it by default. So I need a build flag, something ala:

```
-i18n-warnings          print missing translations
```

As in `print duplicate target paths`.

Suggestions for a flag/config name for such a toggle?

/cc @regisphilibert @kaushalmodi

**Comment by kaushalmodi** [comment]
@bep I'd it's not too late I would make this a config so that one can do:

```toml
[warnings]
  i8n = false
  duplicateTargetPaths = true
```

This would allow adding more warning masks to the same map in future. 

Do you need to have a CLI flag version of this too? If so, it would be nice to have flags like `-warn[NAME]=on/off`. Example: `-warn[i8n]=off`.

**Comment by bep** [comment]
@kaushalmodi this (at least now) a "flag only" thing (I don't want people to have it on all the time). I agree that should do more cleanups on this department, but I think I will have to limit the scope of what I'm doing right now -- else I will never get it out the door.

**Comment by max-arnold** [comment]
What are the downsides of this feature being always on?

**Comment by bep** [comment]
>What are the downsides of this feature being always on?

It creates noise, and a slight performance penalty, but that should be neglible. I have just tested it, and the noise (i.e. stuff you may not bother to fix right away) would be duplicate aliases -- which I suspect is more common than you think.

**Comment by max-arnold** [comment]
I'd suggest to leave it off by default and enable via config option and a command line flag. Another thing to consider is the optional ability to turn these warnings to hard errors (for CI/CD purposes like Netlify builds).
